state_machine/applications/system-check/tests/coil_test.py

```diff
@@ -0,0 +1,130 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+Torque Driver Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from lib import pycubed as cubesat
+from ulab import numpy
+
+# voltage level constants; set between -1 and 1
+v1 = 0.25
+v2 = 0.5
+v3 = 0.75
+
+# find projected voltage: vlevel * max voltage
+projected_voltage1 = v1 * 5.06
+projected_voltage2 = v2 * 5.06
+projected_voltage3 = v3 * 5.06
+
+# alias norm
+norm = numpy.linalg.norm
+
+
+def test_voltage_levels(coil_index):
+    """
+    All user interaction happens in this function
+    Set wait times, prompt user, collect magnetometer data for
+    different voltage levels
+    """
+
+    # wait times
+    wait_time = 3
+    driver_time = 3
+
+    # set up the test
+    print(f'Testing Coil Driver {coil_index} for the following voltage ' +
+          f'levels: {projected_voltage1}, {projected_voltage2}, ' +
+          f'{projected_voltage3}.')
+    print(f"Waiting {wait_time} seconds.")
+    time.sleep(wait_time)
+
+    # conduct the test
+    start_test = input("Type Y to start the test, any key to cancel: ")
+    if start_test.lower() != "y":
+        return None
+
+    # else, test each voltage level
+    cubesat.coildriver_vout(coil_index, projected_voltage1)
+    time.sleep(driver_time)
+    mag_reading1 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage2)
+    time.sleep(driver_time)
+    mag_reading2 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage3)
+    time.sleep(driver_time)
+    mag_reading3 = cubesat.magnetic()
+
+    # calculate total magnetic reading
+    print("Data Collection Complete")
+    mag_total1 = norm(mag_reading1)
+    mag_total2 = norm(mag_reading2)
+    mag_total3 = norm(mag_reading3)
+
+    # if magnetometer readings are increasing over time, return true
+    result_val_bool = (mag_total3 > mag_total2 and mag_total2 > mag_total1)
+    if not result_val_bool:
+        print("Magnetometer results should have been increasing but were" +
+              f" not. reading 1: {mag_total1}, reading 2: {mag_total2}," +
+              f" reading 3: {mag_total3}")
+    return result_val_bool
+
+
+def coil_test(result_dict, coil_index):
+    """
+    All automation happens in this function
```

Comment:
This docstring doesn't really tell me what you are doing in this function.
I think we should change it to something describing the test that is performed